csharp/src/EnumerableExt.cs

```diff
@@ -0,0 +1,107 @@
+﻿// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2022-2022.All rights reserved.
+//   W.Kirschenmann   <wkirschenmann@aneo.fr>
+//   J.Gurhem         <jgurhem@aneo.fr>
+//   D.Dubuc          <ddubuc@aneo.fr>
+//   L.Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F.Lemaitre       <flemaitre@aneo.fr>
+//   S.Djebbar        <sdjebbar@aneo.fr>
+//   J.Fonseca        <jfonseca@aneo.fr>
+//   D.Brasseur       <dbrasseur@aneo.fr>
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using JetBrains.Annotations;
+
+namespace ArmoniK.Utils;
+
+public static class EnumerableExt
+{
+  /// <summary>
+  ///   Convert an enumerable into a list, if it is not already a list
+  ///   Beware that the return list may or may not be a reference to the input enumerable
+  /// </summary>
+  /// <param name="enumerable">The enumerable to convert into a list</param>
+  /// <typeparam name="T">Type of the elements</typeparam>
+  /// <returns>A list containing the same elements as the input enumerable</returns>
+  [PublicAPI]
+  public static IList<T> AsList<T>(this IEnumerable<T>? enumerable)
```

Comment:
rename to `AsIList` because we don't always get a `List<>`

Reply:
I'm not a big of this extra `I`, but OK. Will do.